src/components/timeline/index.js

```diff
@@ -52,21 +59,42 @@ type StateProps = {|
   +panelLayoutGeneration: number,
   +zeroAt: Milliseconds,
   +timelineType: TimelineType,
+  +hiddenTrackCount: HiddenTrackCount,
 |};
 
 type DispatchProps = {|
   +changeGlobalTrackOrder: typeof changeGlobalTrackOrder,
   +commitRange: typeof commitRange,
   +updatePreviewSelection: typeof updatePreviewSelection,
   +changeTimelineType: typeof changeTimelineType,
+  +changeRightClickedTrack: typeof changeRightClickedTrack,
 |};
 
 type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
-class Timeline extends PureComponent<Props> {
+class Timeline extends React.PureComponent<Props> {
   _changeToCategories = () => this.props.changeTimelineType('category');
   _changeToStacks = () => this.props.changeTimelineType('stack');
 
+  _showMenu = (event: SyntheticMouseEvent<HTMLElement>) => {
+    let x = event.clientX;
+    let y = event.clientY;
+    if (event.clientX === 0 && event.clientY === 0) {
+      // This is probably a keyboard event, position the context menu according to
+      // the element's position.
```

Comment:
Actually I'd display it a this position even when we're not using the keyboard, otherwise it feels a bit weird.